integrations/gitlab-v2/gitlab/webhook/webhook_processors/folder_push_webhook_processor.py

```diff
@@ -0,0 +1,77 @@
+from gitlab.webhook.webhook_processors._gitlab_abstract_webhook_processor import (
+    _GitlabAbstractWebhookProcessor,
+)
+from port_ocean.core.handlers.webhook.webhook_event import (
+    EventPayload,
+    WebhookEvent,
+    WebhookEventRawResults,
+)
+from port_ocean.core.handlers.port_app_config.models import ResourceConfig
+from loguru import logger
+from typing import cast
+from integration import GitLabFoldersResourceConfig
+from gitlab.helpers.utils import ObjectKind
+
+
+class FolderPushWebhookProcessor(_GitlabAbstractWebhookProcessor):
+    events = ["push"]
+    hook = "Push Hook"
+
+    async def get_matching_kinds(self, event: WebhookEvent) -> list[str]:
+        return [ObjectKind.FOLDER]
+
+    async def handle_event(
+        self, payload: EventPayload, resource_config: ResourceConfig
+    ) -> WebhookEventRawResults:
+        project_id = payload["project"]["id"]
+        branch = payload.get("ref", "").replace("refs/heads/", "")
+        repo_path = payload["project"]["path_with_namespace"]
+        ref = payload["after"]
+        logger.info(
+            f"Processing push event for project {project_id} on branch {branch} at ref {ref}"
+        )
+
+        config = cast(GitLabFoldersResourceConfig, resource_config)
+        selector = config.selector
+        folder_patterns = selector.folders
+
+        folders = []
+        for pattern in folder_patterns:
+            if not pattern.repos:
```

Comment:
shouldn't repos be "required" and in that case there will not be that kind of if statement?